src/EventHandler/Message.php

```diff
@@ -137,6 +137,73 @@ public function __construct(
         }
     }
 
+    /**
+     * Pin a message
+     *
+     * @param bool $pmOneside Whether the message should only be pinned on the local side of a one-to-one chat
+     * @param bool $silent Pin the message silently, without triggering a notification
+     * @return AbstractMessage|null
+     */
+    public function pin(bool $pmOneside = false,bool $silent = false) : ?AbstractMessage
+    {
+        $result = $this->API->methodCallAsyncRead(
+            'messages.updatePinnedMessage',
+            [
+                'peer' => $this->chatId,
+                'id' => $this->id,
+                'pm_oneside' => $pmOneside,
+                'silent' => $silent,
+                'unpin' => false
+            ]
+        );
+        return $this->API->wrapMessage($this->API->extractMessage($result));
+    }
+
+    /**
+     * Unpin a message
+     *
+     * @param bool $pmOneside Whether the message should only be pinned on the local side of a one-to-one chat
+     * @param bool $silent Pin the message silently, without triggering a notification
+     * @return Update|null
+     */
+    public function unpin(bool $pmOneside = false,bool $silent = false) : ?Update
+    {
+        $result = $this->API->methodCallAsyncRead(
+            'messages.updatePinnedMessage',
+            [
+                'peer' => $this->chatId,
+                'id' => $this->id,
+                'pm_oneside' => $pmOneside,
+                'silent' => $silent,
+                'unpin' => true
+            ]
+        );
+        return $this->API->wrapUpdate($result);
+    }
+
+    /**
+     * React to message
+     *
+     * @param string|array $reaction string or Array of Reaction
+     * @param bool $big Whether a bigger and longer reaction should be shown
+     * @param bool $addToRecent Add this reaction to the recent reactions list.
+     * @return Update|null
+     */
+    public function react(string|array $reaction,bool $big = false,bool $addToRecent = true) : ?Update
```

Comment:
This should be 3 methods:
- getReactions
- addReaction
- delReaction


All of which should return and take either strings (for normal emojis) or integers (for custom emojis).
getReactions should return an array of strings or integers, stored in a local `reactions` property, add/del should make method calls and update the local property on success.
The `reactions` property should be populated from the `reactions` field of the message (this is a bit more complicated, I think I should give a go at this).

Reply:
getReactions its ok and i can implement it!
but about addReaction and delReaction reaction you should pay attention that call of method react on the same reaction will delete the reaction so i think there is no need to implement 2 methods for that ...